### PR TITLE
 Linter to avoid use of db_columns in Model fields

### LINKED ISSUE #736: new linter: do not use db_column in model fields
`db_column` argument in model field definitions is a legacy leftover and in many cases the column name in the DB is a bit different from the model field name. All of this has been fixed but it took lots of effort to do so.

Kiwi TCMS needs a custom kiwi_lint/ checker which will discover all of these issues (if we forgot to fix some) and prevent them from happening again.

**Definition of Done**:
- new checker class under `kiwi_lint/` which implements the necessary pylint machinery
- Can be relatively simple checker b/c the `db_column` parameter name is rather unique and we probably don't use it for anything not related to declaring Django models. It should be sufficient to raise a warning when a function call contains a parameter with the name db_column. 
- All code review passed & PR merged

Here's me explaining how to write pylint plugins:
- https://www.youtube.com/watch?v=3CkSKUNMLJc
- https://www.youtube.com/watch?v=D1wLFyazTRM

*This issue is part of Kiwi TCMS open source bounty program. For more information see the link(s) in [bounty-program milestone](https://github.com/kiwitcms/Kiwi/milestone/4)*

### PR DESCRIPTION
**Pull request by souravsingh**
Attempt to fix #736. The linter would probably need improvements

### CODE REVIEW

File: kiwi_lint/db_columns.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from pylint import interfaces, checkers
```
Comment (atodorov) [inline kiwi_lint/db_columns.py:1]: Linter needs to be registered in `__init__.py`

File: kiwi_lint/db_columns.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+from pylint.checkers import utils
+
+
+class AutoFieldChecker(checkers.BaseChecker):
```
Comment (atodorov) [inline kiwi_lint/db_columns.py:5]: Naming ! This has been copy pasted

File: kiwi_lint/db_columns.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    __implements__ = (interfaces.IAstroidChecker, )
+
+    name = 'db-columns-checker'
```
Comment (atodorov) [inline kiwi_lint/db_columns.py:8]: Don't mix singular and plural in names. The attribute under interest is `db_column` so everything should be named after it.

File: kiwi_lint/db_columns.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    msgs = {
+        'E4480': (
```
Comment (atodorov) [inline kiwi_lint/db_columns.py:11]: 4811, order is from top to bottom in the way checkers are registered in `__init__.py`

File: kiwi_lint/db_columns.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+            'Avoid using db_columns in Model fields',
+            'avoid-db-columns'
+        )
+    }
+
+    @utils.check_messages('avoid-db-column')
+    def visit_attribute(self, node):
+        if node.attrname == 'db_column':
+            self.add_message('avoid-db-columns', node=node)
```
Comment (atodorov) [inline kiwi_lint/db_columns.py:20]: This visit method is not correct. 

```
>>> n = astroid.parse("""name = models.CharField(db_column='tag_name', max_length=255)""")
>>> print(n.repr_tree())
Module(
   name='',
   doc=None,
   file='<?>',
   path=['<?>'],
   package=False,
   pure_python=True,
   future_imports=set(),
   body=[Assign(
         targets=[AssignName(name='name')],
         value=Call(
            func=Attribute(
               attrname='CharField',
               expr=Name(name='models')),
            args=[],
            keywords=[Keyword(
                  arg='db_column',
                  value=Const(value='tag_name')),
               Keyword(
                  arg='max_length',
                  value=Const(value=255))]))])
```

File: kiwi_lint/db_columns.py (hunk 1 above)
Comment (atodorov) [reply kiwi_lint/db_columns.py:1]: ^^^

File: kiwi_lint/db_columns.py (hunk 4 above)
Comment (atodorov) [reply kiwi_lint/db_columns.py]: ^^^

File: kiwi_lint/db_columns.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        'E4481': (
+            'Avoid using db_column in Model fields',
+            'avoid-db-column'
+        )
+    }
+
+    @utils.check_messages('avoid-db-column')
+    def visit_attribute(self, node):
+        if (node.attrname == 'CharField'
+                and isinstance(node.expr, astroid.Attribute) and node.expr.attrname == 'models'):
```
Comment (atodorov) [inline kiwi_lint/db_columns.py:20]: still incorrect.

Comment (souravsingh) [reply kiwi_lint/db_columns.py:20]: Do we use the `node.func.as_string().endswith('Field')` to check for the node?

Comment (atodorov) [reply kiwi_lint/db_columns.py:20]: All field types can have db_column attribute, not only CharField. Try pylint_django.utils.node_is_subclass

File: kiwi_lint/__init__.py [hunk 7]
```diff
@@ -40,3 +41,5 @@ def register(linter):
     linter.register_checker(OneToOneFieldChecker(linter))
     linter.register_checker(ClassBasedViewChecker(linter))
     linter.register_checker(DatetimeChecker(linter))
+    linter.register_checker(DbColumnChecker(linter))
```
Comment (atodorov) [inline kiwi_lint/__init__.py:44]: Rebase to latest master and register after `ModelFormChecker`

File: kiwi_lint/__init__.py [hunk 8]
```diff
@@ -40,3 +41,5 @@ def register(linter):
...
+
```
Comment (atodorov) [inline kiwi_lint/__init__.py:45]: no need for empty line, remove it

File: kiwi_lint/db_columns.py (hunk 6 above)
Comment (atodorov) [reply kiwi_lint/db_columns.py:20]: This is still unresolved.

**Comment by atodorov** [comment]
Closing in favor of https://github.com/kiwitcms/Kiwi/pull/1887

**Pull request closed without merging**